scripts/ci/lib.sh

```diff
@@ -470,7 +470,7 @@ mark_collector_release() {
     info "Create a branch for the PR"
 
     collector_version="$(cat COLLECTOR_VERSION)"
-    cd /tmp/collector || exit
+    pushd /tmp/collector || exit
```

Comment:
If I am not mistaken, this was refactored from a stand-alone script into a bash function; and `cd`s from the script invocation were obviously not propagated back to the caller. Not sure why this refactoring was done, but I think a better way to eliminate the problems introduced by it would be to have the entire function spawn a subshell. I.e., instead of writing `mark_collector_release() { ... }`, do `mark_collector_release() ( ... )`. This ensures the invoking shell will no longer have its state modified by the function, without depending on the user to revert modified state manually on every exit path.

Make sure that the function is properly invoked with `set -e` set or with a `... || die`, because `die` in l.459 and `exit` here will no longer propagate either.

Reply:
>Not sure why this refactoring was done

I think it was easier to migrate to OSCI that way (provided the time pressure). Gavin would know more about this.

> I think a better way to eliminate the problems introduced by it would be to have the entire function spawn a subshell

We can try this, as this would nicely isolate things. Alternatively, we could also operate with absolute paths without depending on `cd`. I think sub-shell may be cleaner from those two options.